File: cpp/factorization.cpp
vector<pair<long long,int>> factorization(long long x){
    vector<pair<long long,int>> resu;
    for(long long i = 2;i*i <= x; ++i){
        if(x%i)continue;
        int e = 0;
        while(x%i == 0){
            ++e;
            x /= i;
        }
        resu.push_back({i, e});
    }

    if(x != 1){
        resu.push_back({x, 1});
    }

    return resu;
}

File: rust/factorization.rs
fn factorization(mut x: i64) -> Vec<(i64, i64)> {
    let mut resu = Vec::new();
    for i in 2.. {
        if i*i > x { break; }
        if x%i != 0{ continue; }

        let mut e = 0;
        while x%i == 0 {
            e += 1;
            x /= i;
        }

        resu.push((i, e));
    }

    if x != 1 {
        resu.push((x, 1));
    }

    resu
}